src/index.js

```diff
@@ -9,7 +10,17 @@ const startServer = async () => {
 
     // Running the application in test mode does not start listening because parallel tests would result in EADDRINUSE
     if (process.env.NODE_ENV !== "test") {
-        app.listen(config.port, (err) => {
+
+        let server = app;
+        if (process.env.NODE_ENV !== "production") {
+            const fs = require("fs");
+            const path = require("path");
+            const key = fs.readFileSync(path.join(__dirname, "../certs/key.pem"));
+            const cert = fs.readFileSync(path.join(__dirname, "../certs/cert.pem"));
```

Comment:
This is not a remark as much as a doubt really, but why does the certificates retrieval need to be synchronous?

Reply:
It has to be synchronous, since it's needed to start the server, but it's effectively the same as using await on the Promise version of fs. Actually, I can make these 2 run concurrently by having something like:

```js
const [key, cert] = await Promise.all([
    fs.promises.readFile(path.join(__dirname, "../certs/key.pem")),
    fs.promises.readFile(path.join(__dirname, "../certs/cert.pem"))
]);
```

Thanks for the callout!